test/langtools/jdk/javadoc/doclet/testSnippetTag/TestSnippetTag.java

```diff
@@ -0,0 +1,1878 @@
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 8201533
+ * @library /tools/lib ../../lib
+ * @modules jdk.compiler/com.sun.tools.javac.api
+ *          jdk.compiler/com.sun.tools.javac.main
+ *          jdk.javadoc/jdk.javadoc.internal.tool
+ * @build javadoc.tester.* toolbox.ToolBox toolbox.ModuleBuilder builder.ClassBuilder
+ * @run main TestSnippetTag
+ */
+
+import builder.ClassBuilder;
+import builder.ClassBuilder.MethodBuilder;
+import javadoc.tester.JavadocTester;
+import toolbox.ModuleBuilder;
+import toolbox.ToolBox;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardOpenOption;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.function.ObjIntConsumer;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Stream;
+
+// FIXME
+//   0. Add tests for snippets in all types of elements: e.g., fields
+//      and constructors (i.e. not only methods.)
+//   1. Add tests for nested structure under "snippet-files/"
+//   2. Add negative tests for region
+//   3. Add tests for hybrid snippets
+
+/*
+ * General notes.
+ *
+ * 1. Some of the below tests could benefit from using a combinatorics library
+ * as they are otherwise very wordy.
+ *
+ * 2. One has to be careful when using JavadocTester.checkOutput with duplicating
+ * strings. If JavadocTester.checkOutput(x) is true, then it will also be true
+ * if x is passed to that method additionally N times: JavadocTester.checkOutput(x, x, ..., x).
+ * This is because a single occurrence of x in the output will be matched N times.
+ */
+public class TestSnippetTag extends JavadocTester {
+
+    private final ToolBox tb = new ToolBox();
+
+    private TestSnippetTag() { }
+
+    public static void main(String... args) throws Exception {
+        new TestSnippetTag().runTests(m -> new Object[]{Paths.get(m.getName())});
```

Comment:
This is just a suggestion for future convenience.
For big tests like this, for ease of debugging, it can be useful to be able to specify the method(s) to be executed.   Maybe there should be a variants of `runTests` that takes a `String[]` or `List<String>`

Reply:
I filed JDK-8272853 for us.